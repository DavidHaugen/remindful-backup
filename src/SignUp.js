import React, { Component } from 'react'

class SignUp extends Component {
  render(){
    return(
      <div> 
        <h2>Welcome to Remindful</h2>
        <h3>Please enter your email address and password to get started</h3>
        <h4>We will use this email address to send you a reminder to check in and reflect on your progress on the first of every month</h4>
        <form>
          <div>
            <label>First name
              <input />
            </label>
          </div>
          <div>
            <label>Last name
              <input />
            </label>
          </div>
          <div>
            <label>Email address
              <input />
            </label>
          </div>
          <div>
            <label>Password
              <input type='password'/>
            </label>
          </div>
          <div>
            <label>Confirm password
              <input type='password'/>
            </label>
          </div>
          <button type='submit'>Submit</button>
        </form>
      </div>    
    )
  }
}

export default SignUp