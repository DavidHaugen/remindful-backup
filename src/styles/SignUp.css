*{
  font-family: 'Rubik', sans-serif;
}

.aboutLink{
  font-family: 'Rubik', sans-serif;
  font-size: 20px;
  text-decoration: none;
  color: #1c2a43;
}