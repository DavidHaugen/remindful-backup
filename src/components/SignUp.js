import React, { Component } from 'react'
import AuthApiService from '../services/auth-api-service'
import { Link } from 'react-router-dom'
import '../styles/SignUp.css'

class SignUp extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { first_name, last_name, email_address, password } = ev.target
    this.setState({ error: null })
    AuthApiService.postUser({
      email_address: email_address.value,
      password: password.value,
      first_name: first_name.value,
      last_name: last_name.value,
    })
      .then(() => {
        first_name.value = ''
        last_name.value = ''
        email_address.value = ''
        password.value = ''
        this.props.history.push('/my-goals')
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render(){
    const { error } = this.state

    return(
      <div className="main"> 
        <div className="wrapper">
          <h1 className="karla">Welcome to Remindful</h1>
          <p className="instructions">Please enter your email address and password to get started</p>
          <p className="instructions">We will use this email address to send you a reminder to check in and reflect on your progress on the first of every month</p>
          <div className="formContainer">
            <form onSubmit={this.handleSubmit} className="inputForm">
              <div role='alert'>
                {error && <p className='red'>{error}</p>}
              </div>
              <div className='formField'>
                <label className="inputLabel" htmlFor="signUp_first_name">First name</label>
                  <input name='first_name' type='text' id='signUp_first_name' className="inputField" required />
              </div>
              <div className='formField'>
                <label className="inputLabel" htmlFor="signUp_last_name">Last name</label>
                  <input name='last_name' type='text' id='signUp_last_name' className="inputField" required/>
              </div>
              <div className='formField'>
                <label className="inputLabel" htmlFor="signUp_email_address">Email address</label>
                  <input name='email_address' type='email' id='signUp_email_address' className="inputField" required/>
              </div>
              <div className='formField'>
                <label className="inputLabel" htmlFor="password">Password</label>
                  <input name='password' type='password' id='password' className="inputField" required/>
              </div>
              <button type='submit' className="textButton login">Sign up</button>
            </form>
          </div>
          <section className="instructions">
            <p>Want to try Remindful before signing up? See our {<Link className='aboutLink' to='/'>About</Link>} page!</p>
          </section>
        </div>
      </div>    
    )
  }
}

export default SignUp