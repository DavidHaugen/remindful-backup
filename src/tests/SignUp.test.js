import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import SignUp from '../components/SignUp'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer'

describe('<SignUp />', () => {
  it('Renders without crashing', () => {
      mount(<MemoryRouter><SignUp/></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><SignUp/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});