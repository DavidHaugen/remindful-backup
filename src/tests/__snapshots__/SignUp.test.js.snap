// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders the UI as expected 1`] = `
<div
  className="main"
>
  <div
    className="wrapper"
  >
    <h1
      className="karla"
    >
      Welcome to Remindful
    </h1>
    <p
      className="instructions"
    >
      Please enter your email address and password to get started
    </p>
    <p
      className="instructions"
    >
      We will use this email address to send you a reminder to check in and reflect on your progress on the first of every month
    </p>
    <div
      className="formContainer"
    >
      <form
        className="inputForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="signUp_first_name"
          >
            First name
          </label>
          <input
            className="inputField"
            id="signUp_first_name"
            name="first_name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="signUp_last_name"
          >
            Last name
          </label>
          <input
            className="inputField"
            id="signUp_last_name"
            name="last_name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="signUp_email_address"
          >
            Email
          </label>
          <input
            className="inputField"
            id="signUp_email_address"
            name="email_address"
            required={true}
            type="email"
          />
        </div>
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="inputField"
            id="password"
            name="password"
            required={true}
            type="password"
          />
        </div>
        <button
          className="textButton login"
          type="submit"
        >
          Sign up
        </button>
      </form>
    </div>
    <section
      className="instructions"
    >
      <p>
        Want to try Remindful before signing up? See our 
        <a
          className="aboutLink"
          href="/about"
          onClick={[Function]}
        >
          About page!
        </a>
      </p>
    </section>
  </div>
</div>
`;
